stix2/base.py

```diff
@@ -163,6 +171,13 @@ def __init__(self, allow_custom=False, **kwargs):
                 raise ExtraPropertiesError(cls, extra_kwargs)
         if custom_props:
             self._allow_custom = True
+            if self.get_class_version() == "v21":
+                for prop_name, prop_value in custom_props.items():
+                    if not re.match(PREFIX_21_REGEX, prop_name):
+                        raise InvalidValueError(
+                            self.__class__, prop_name,
+                            reason="Property name '%s' must begin with an alpha character." % prop_name,
+                        )
```

Comment:
I think this only checks custom properties when the object was created with the `custom_properties` parameter.  You can also give the property directly and include `allow_custom=True`, and then the property name wouldn't be checked.  So I think this check needs to be done in a different way.  For example:

```python
f = stix2.v21.File(name="foo.exe", _a_b_=11, allow_custom=True)
```

produces no error.  But:

```python
f = stix2.v21.File(name="foo.exe", custom_properties={"_a_b_": 11})
```

produces an error.

Reply:
I was more concerned about testing with a leading numeric character - which would be impossible to do using the first call above.  I wonder if this should be an illegal property name anyway.

Reply:
Yes, we need to enforce this with `allow_custom=True` too.